.github/workflows/sam-automation-tests.yaml

```
@@ -0,0 +1,113 @@
+name: sam-automation-tests
+
+on:
+  pull_request:
+    paths-ignore: ['**.md']
+  push:
+    branches:
+      - develop
+    paths-ignore: ['**.md']
+
+jobs:
+  sam-build-tag-publish-job:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    outputs:
+      custom-version-json: ${{ steps.render-data.outputs.custom-version-json }}
+    steps:
+      - name: Extract branch
+        id: extract-branch
+        run: |
+          GITHUB_EVENT_NAME=${{ github.event_name }}
+          if [[ "$GITHUB_EVENT_NAME" == "push" ]]; then
+            GITHUB_REF=${{ github.ref }}
+            GITHUB_SHA=${{ github.sha }}
+          elif [[ "$GITHUB_EVENT_NAME" == "pull_request" ]]; then
+            GITHUB_REF=refs/heads/${{ github.head_ref }}
+            GITHUB_SHA=${{ github.event.pull_request.head.sha }}
+          else
+            echo "Failed to extract branch information"
+            exit 1
+          fi
+
+          echo ::set-output name=ref::$GITHUB_REF
+          echo ::set-output name=sha::$GITHUB_SHA
+
+      - name: dispatch build to terra-github-workflows
+        uses: aurelien-baudet/workflow-dispatch@93e95b157d791ae7f42aef8f8a0d3d723eba1c31 #commit sha for v2.1.1
+        with:
+          workflow: sam-build
+          repo: broadinstitute/terra-github-workflows
+          ref: refs/heads/main
+          token: ${{ secrets.BROADBOT_TOKEN}} # github token for access to kick off a job in the private repo
+          inputs: '{ "repository": "${{ github.event.repository.full_name }}", "ref": "${{ steps.extract-branch.outputs.ref }}" }'
+
+      - name: Render data to share with dependent jobs
+        id: render-data
+        env:
+          GITHUB_CONTEXT: ${{ toJSON(github) }}
+        run: |
+          echo "$GITHUB_CONTEXT"
+          GIT_SHA=${{ steps.extract-branch.outputs.sha }}
+          HASH_TAG=${GIT_SHA:0:12}
+          # Need to escape twice because custom-version-json is passed like an cli argument string
+          echo "HASH_TAG=$HASH_TAG"
+          echo ::set-output name=custom-version-json::"{\\\"sam\\\":{\\\"appVersion\\\":\\\"$HASH_TAG\\\"}}"
+
+  create-bee-workflow:
+    runs-on: ubuntu-latest
+    needs: [sam-build-tag-publish-job]
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - name: Check hash tag
```

nit:
```suggestion
      - name: Print hash tag
```
Since it's not actually doing any checking/verifying

---

Updated comment. Rerun tests. All 12 checks passed.